### PR TITLE
feat: trades from stream improvements

### PR DESCRIPTION
**Pull request by TomMcL**
### Description

Adding a filter to trades_from_stream to avoid trades you don't want being included in the return. Also freezing data class such that it can be returned without copying.

Additional various other improvements to streaming and caching, mainly focused around keeping market sim performant under high numbers of repeated trading sessions. RL runs now average 2s out to 100+ iterations. After this point they start to slow, so also added an automatic restart of Vega at this point, which resets the slowdown and allows 2s runs continuously (by this point the training time in current models outweighs the run time significantly, so further improvement is probably not too worthwhile)

image

### Testing

Tests pass

### Breaking Changes

Should be none

### Closes

#304

### CODE REVIEW

**Review by cdummett — DISMISSED** [review]
LGTM

**Review by davidsiska-vega — CHANGES_REQUESTED** [review]
Looks good - my only thought is - do we want to hold off till the issue with the RL nightly runs is resolved? 

Or maybe let’s make the restart 100 iterations optional and not do that on the RL nightly runs (at least for now).

**Review by davidsiska-vega — DISMISSED** [review]

File: vega_sim/service.py [hunk 1]
```diff
@@ -154,6 +157,9 @@ def __init__(
         self._aggregated_observation_feed: Queue[Any] = Queue()
         self._kill_thread_sig = threading.Event()
 
+        self._observation_feeds: List[Queue[Any]] = []
+        self._aggregated_observation_feed: Queue[Any] = Queue()
+
```
Comment (cdummett) [inline vega_sim/service.py]: Duplicated?

**Review by cdummett — CHANGES_REQUESTED** [review]
Having an issue when running a scenario I am hanging after settling the market - means all my local tests are hanging as well.

Seems fine in the develop branch.

**Comment by TomMcL** [comment]
@cdummett thanks for catching, looks like the kill signal got lost in the merge! Should be working again now 👍

**Review by cdummett — APPROVED** [review]

**Pull request merged**